lightning/src/ln/channel.rs

```diff
@@ -10876,8 +10911,49 @@ where
 			}
 		}
 
-		let prev_funding_input = self.funding.to_splice_funding_input();
+		let original_funding_txo = self.funding.get_funding_txo().ok_or_else(|| {
+			debug_assert!(false);
+			APIError::APIMisuseError {
+				err: "Chanel isn't yet fully funded".to_owned(),
+			}
+		})?;
+
 		let (our_funding_inputs, our_funding_outputs, change_script) = contribution.into_tx_parts();
+
+		let action = QuiescentAction::Splice(SpliceInstructions {
+			adjusted_funding_contribution,
+			our_funding_inputs,
+			our_funding_outputs,
+			change_script,
+			funding_feerate_per_kw,
+			locktime,
+			original_funding_txo,
+		});
+		self.propose_quiescence(logger, action)
+			.map_err(|e| APIError::APIMisuseError { err: e.to_owned() })
```

Comment:
Would it make sense to break the check portion out of `propose_quiescence` such that it can be checked earlier?